docs/booster.md

```diff
@@ -0,0 +1,195 @@
+│         │                   │                               │                                 │                                                │  │
+│         │                   │                               │                                 │                                                │  │
+│         ▼                   ▼                               ▼                                 ▼                                                │  │
+│                                                                                                                                                │  │
+│  Return aborted       No rules apply                 Rewrite to P'  ───┐                  Rewrite to PS ─────────────────┬───────┐             │  │
+│                                                                        │                                                 │       │             │  │
+│                             │                           │              │                    │      │                     │       │             │  │
+│                             │                           │              │                    │      └──────────┐          │       │             │  │
+│                             │                           ▼              ▼                    ▼                 ▼          │       ▼             │  │
+│                             │                                                                                            │                     │  │
+│                             │                         P' == _|_    P' /= _|_           /\ PS == _|_      PS simplify to  │   PS simplify to  ──┘  │
+│                             │                                                                                   []       │      single P'         │
+│              ┌──────────────┼─────────────┐               │           │ │                   │                            │                        │
+│              │              │             │               │           │ │                   │                   │        │                        │
+│              ▼              ▼             ▼               │           │ │                   │                   │        └───────┐                │
+│                                                           │           │ │                   │                   │                ▼                │
+│          Does not     Simplified      Simplifies          │  ┌────────┼─┼───────────────────┘                   │                                 │
+│          simplify      already                            │  │        │ │                                       │          PS simplify to         │
+│                                                           │  │        │ │                                       │                PS'              │
```

Comment:
What's the difference between "Does not simplify" and "Simplified already"?

Reply:
Simplified already means that we have done a loop where no rules applied, then we were able to simplify the configuration and tried rule application again, which produced no results. Does not simplify just means we took a rewrite step, then tried to take another one but no applicable rules were found, then we tried to simplify but the configuration wasn't simplified so we return stuck immediately.

I think in the diagram, we can remove this distinction, because functionally, the simplified diagram should be the same